roles/openshift_health_checker/openshift_checks/diskspace_availability.py

```diff
@@ -0,0 +1,94 @@
+# pylint: disable=missing-docstring
+from openshift_checks import OpenShiftCheck, get_var
+
+
+class DiskAvailability(OpenShiftCheck):
+    """Check that recommended disk space is available."""
+
+    name = "disk_availability"
+    tags = ["preflight"]
+
+    recommended_node_space_gb = 15.0
+    recommended_master_space_gb = 40.0
+    minimum_docker_space_percent = 5.0
+
+    supported_docker_dev_driver = "devicemapper"
+
+    def run(self, tmp, task_vars):
+        is_containerized = get_var(task_vars, "openshift", "common", "is_containerized")
+        group_names = get_var(task_vars, "group_names", default=[])
+        ansible_mounts = get_var(task_vars, "ansible_mounts")
+        diskfree = self.to_gibibytes(self.get_disk_space(ansible_mounts))
+
+        # if running containerized installation check
+        # that available space is not below 5% instead
+        if is_containerized:
+            docker_facts = self.module_executor("docker_facts", {}, task_vars)
+            info = docker_facts.get("docker_facts")
+            status = dict(info.get("DriverStatus"))
+
+            if info.get("Driver", None) == self.supported_docker_dev_driver:
+                disktotal = status.get("Data Space Total", None)
+                diskavail = status.get("Data Space Available", None)
+                diskleft_percent = self.to_bytes(diskavail) / self.to_bytes(disktotal) * 100
+
+                # fail if less than 5% of available dataspace space left
+                if int(diskleft_percent) < self.minimum_docker_space_percent:
+                    msg = "The amount of data space remaining (%s%%) for the docker storage driver is below %s%%" \
+                          % (int(diskleft_percent), self.minimum_docker_space_percent)
+                    return {"failed": True, "msg": msg}
+
+            return {"changed": False}
+
+        recommended_diskspace_gb = self.recommended_master_space_gb
+
+        if "nodes" in group_names:
+            recommended_diskspace_gb = self.recommended_node_space_gb
+
+        if float(diskfree) < recommended_diskspace_gb:
+            msg = "Available disk space for %s (%s) below recommended storage. Minimum required disk space: %s GB" \
+                  % ("node" if "nodes" in group_names else "master", str(diskfree) + " GB", recommended_diskspace_gb)
+            return {"failed": True, "msg": msg}
+
+        return {"changed": False}
+
+    @staticmethod
+    def get_disk_space(ansible_mounts):
+        if not ansible_mounts:
+            return None
+
+        root_size_available = None
+        for mnt in ansible_mounts:
+            if mnt.get("mount") == "/":
+                root_size_available = mnt.get("size_available")
+            if mnt.get("mount") == "/var":
+                return mnt.get("size_available")
```

Comment:
I return immediately if `/var` exists at all. If not, `/` is the fallback mount path. I update this method to be a bit more legible